includes/payment-methods/class-wc-gateway-youcanpay-standalone.php

```diff
@@ -1,238 +1,247 @@
-		$return_url            = $this->get_youcanpay_return_url( $order, self::ID );
-		$post_data             = [];
-		$post_data['amount']   = WC_YouCanPay_Helper::get_youcanpay_amount( $order->get_total(), $currency );
-		$post_data['currency'] = strtoupper( $currency );
-		$post_data['type']     = 'standalone';
-		$post_data['owner']    = $this->get_owner_details( $order );
-		$post_data['redirect'] = [ 'return_url' => $return_url ];
-
-		WC_YouCanPay_Logger::info( 'Info: Begin creating YouCan Pay Standalone source' );
-
-		return WC_YouCanPay_API::request( $order, $post_data );
-	}
-
-	/**
-	 * Process the payment
-	 *
-	 * @param int $order_id Reference.
-	 *
-	 * @return array
-	 */
-	public function process_payment( $order_id ) {
-		try {
-			$order = wc_get_order( $order_id );
-
-			// This will throw exception if not valid.
-			$this->validate_minimum_order_amount( $order );
-
-			$response = $this->create_source( $order );
-
-			if ( ! empty( $response->error ) ) {
-				$order->add_order_note( $response->error->message );
-
-				throw new WC_YouCanPay_Exception( print_r( $response, true ), $response->error->message );
-			}
-
-			$order->update_meta_data( '_youcanpay_source_id', $response->id );
-			$order->save();
-
-			WC_YouCanPay_Logger::info( 'Info: Redirecting to YouCan Pay Standalone...' );
-
-			return [
-				'result'   => 'success',
-				'redirect' => esc_url_raw( $response->redirect->url ),
-			];
-		} catch ( WC_YouCanPay_Exception $e ) {
-			wc_add_notice( $e->getLocalizedMessage(), 'error' );
-			WC_YouCanPay_Logger::alert( 'wc youcan pay exception', array(
-				'exception.message' => $e->getMessage()
-			) );
-
-			$statuses = apply_filters(
-				'wc_youcanpay_allowed_payment_processing_statuses',
-				[ 'pending', 'failed' ]
-			);
-
-			if ( isset( $order ) ) {
-				$order->update_status( 'failed' );
-				if ( $order->has_status( $statuses ) ) {
-					$this->send_failed_order_email( $order_id );
-				}
-			}
-
-			return [
-				'result'   => 'fail',
-				'redirect' => '',
-			];
-		}
-	}
+    }
+
+    /**
+     * Creates the source for charge.
+     *
+     * @param WC_Order|WC_Order_Refund $order
+     *
+     * @return array|stdClass
+     */
+    public function create_source($order)
+    {
+        $currency = $order->get_currency();
+        $return_url = $this->get_youcanpay_return_url($order, self::ID);
+        $post_data = [
+            'amount' => WC_YouCanPay_Helper::get_youcanpay_amount($order->get_total(), $currency),
+            'locale' => WC_YouCanPay_Helper::get_supported_local(get_locale()),
+            'currency' => strtoupper($currency),
+            'type' => 'standalone',
+            'owner' => $this->get_owner_details($order),
+            'redirect' => ['return_url' => $return_url],
+        ];
+
+        return WC_YouCanPay_API::request($order, $post_data);
+    }
+
+    /**
+     * Process the payment
+     *
+     * @param int $order_id Reference.
```

Comment:
for the the phpdoc should contain exceptions same for other phpdocs 🙏

Reply:
@AmineAlyate Why? can you please explain more? we always have only array response

Reply:
https://github.com/NextmediaMa/youcan-pay-for-woocommerce/pull/12/commits/8f619601e2f3498862395a06dff350c1928dc177